plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php

```diff
@@ -145,4 +160,90 @@ private function preload_poster_image( string $poster, OD_Tag_Visitor_Context $c
 			);
 		}
 	}
+
+	/**
+	 * Optimizes the VIDEO tag based on whether it is the LCP element or else whether it is displayed in any initial viewport.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param non-empty-string|null  $poster  Poster image URL.
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context, with the cursor currently at an embed block.
+	 */
+	private function lazy_load_videos( ?string $poster, OD_Tag_Visitor_Context $context ): void {
+		$processor = $context->processor;
+
+		/*
+		 * Do not do any lazy-loading if the mobile and desktop viewport groups lack URL metrics. This is important
+		 * because if there is a VIDEO in the initial viewport on desktop but not mobile, if then there are only URL
+		 * metrics collected for mobile then the VIDEO will get lazy-loaded which is good for mobile but for desktop
+		 * it will hurt performance. So this is why it is important to have URL metrics collected for both desktop and
+		 * mobile to verify whether maximum intersectionRatio is accounting for both screen sizes.
+		 * TODO: Add this same condition to IMG lazy-loading and Embed lazy-loading.
```

Comment:
This has been implemented in the follow-up PR: https://github.com/WordPress/performance/pull/1604